cpp/CMakeLists.txt

```diff
@@ -194,7 +194,7 @@ target_link_libraries(cudf rmm "${ARROW_LIB}")
 add_custom_command(OUTPUT PYTHON_CFFI
                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
                    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_BINARY_DIR}/../python ${CMAKE_BINARY_DIR}/python
-                   COMMAND ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR}/python python setup.py build_ext --inplace
+                   COMMAND ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR}/python python setup.py install
```

Comment:
I'm not comfortable with this change. This makes it so that the only way to build the cffi is to also install it. I don't make install anything when I'm developing from source (leads to a pile of old versions in your site_packages which eventually leads to confusing and hard-to-diagnose errors).  We need to have install and build as separate cmake commands.

Reply:
The way to install is with the below command make install_python (which you use correctly in the Dockerfile).

Reply:
What I don't understand is why it's not working, since the necessary calls were already there.

Reply:
The fix was to `conda install cffi` prior to `make install_python`. I backed out all the changes to CMakeLists.txt